integration-tests/smoke/ccip/ccip_token_transfer_test.go

```diff
@@ -217,5 +360,182 @@ func TestTokenTransfer(t *testing.T) {
+	require.NoError(t, err)
+
+	externalTokenPoolsSignerPDA, _, err := solstate.FindExternalTokenPoolsSignerPDA(state.SolChains[sourceChain].Router)
+	require.NoError(t, err)
+
+	ixApprove2, err := soltokens.TokenApproveChecked(1000, 9, solana.Token2022ProgramID, userTokenAccount, srcToken, externalTokenPoolsSignerPDA, ownerSourceChain, nil)
+	require.NoError(t, err)
+
+	ixs := []solana.Instruction{ixApprove2}
+	result := soltestutils.SendAndConfirm(ctx, t, rpcClient, ixs, *sender, solconfig.DefaultCommitment)
+	require.NotNil(t, result)
+	// END: extract as MintAndAllow on Solana
+
+	// ---
+	emptyEVMExtraArgsV2 := []byte{}
+	extraArgs := emptyEVMExtraArgsV2
+
+	// extraArgs := soltestutils.MustSerializeExtraArgs(t, fee_quoter.EVMExtraArgsV2{
+	// 	GasLimit: bin.Uint128{Lo: 500_000, Hi: 0}, // TODO: why is default not enough
+	// }, solccip.EVMExtraArgsV2Tag)
+
+	tcs := []testhelpers.TestTransferRequest{
+		{
+			Name:        "Send token to contract",
+			SourceChain: sourceChain,
+			DestChain:   destChain,
+			SolTokens: []ccip_router.SVMTokenAmount{
+				{
+					Token:  srcToken,
+					Amount: 1,
+				},
+			},
+			Receiver: state.Chains[destChain].Receiver.Address().Bytes(),
+			ExpectedTokenBalances: []testhelpers.ExpectedBalance{
+				// due to the differences in decimals, 1 on SVM results to 1e9 on EVM
+				{Token: destToken.Address().Bytes(), Amount: new(big.Int).SetUint64(oneE9)},
+			},
+			ExtraArgs:      extraArgs,
+			ExpectedStatus: testhelpers.EXECUTION_STATE_SUCCESS,
+		},
+		// {
+		// 	Name:        "Send N tokens to contract",
+		// 	SourceChain: destChain,
+		// 	DestChain:   sourceChain,
+		// 	Tokens: []router.ClientEVMTokenAmount{
+		// 		{
+		// 			Token:  selfServeDestToken.Address(),
+		// 			Amount: oneE9,
+		// 		},
+		// 		{
+		// 			Token:  destToken.Address(),
+		// 			Amount: oneE9,
+		// 		},
+		// 		{
+		// 			Token:  selfServeDestToken.Address(),
+		// 			Amount: oneE9,
+		// 		},
+		// 	},
+		// 	Receiver:  state.Chains[sourceChain].Receiver.Address().Bytes(),
+		// 	ExtraArgs: testhelpers.MakeEVMExtraArgsV2(300_000, false),
+		// 	ExpectedTokenBalances: []testhelpers.ExpectedBalance{
+		// 		{selfServeSrcToken.Address().Bytes(), new(big.Int).Add(oneE18, oneE18)},
+		// 		{srcToken.Address().Bytes(), oneE18},
+		// 	},
+		// 	ExpectedStatus: testhelpers.EXECUTION_STATE_SUCCESS,
+		// },
+	}
+
+	startBlocks, expectedSeqNums, _, expectedTokenBalances :=
+		testhelpers.TransferMultiple(ctx, t, e, state, tcs)
+
+	err = testhelpers.ConfirmMultipleCommits(
+		t,
+		e,
+		state,
+		startBlocks,
+		false,
+		expectedSeqNums,
+	)
+	require.NoError(t, err)
+
+	_ = testhelpers.ConfirmExecWithSeqNrsForAll(
+		t,
+		e,
+		state,
+		testhelpers.SeqNumberRangeToSlice(expectedSeqNums),
+		startBlocks,
+	)
+	// NOTE: can't validate this since there will be more execution states due to price updates being separate
```

Comment:
What does ExecutionStateChanged have to do with price updates? Aren't those only emitted on msg executions?

Reply:
Hmm I think my comment is wrong here but for some reason the sequence was higher than `1` here for me (the tests for EVM assert seqNum = old num + 1). I think commit reports with price only updates still end up raising the seq number, so the total number was higher for me

Reply:
Hmm it seems to pass for me again so I'll re-enable